src/components/upload/useUpload.ts

```diff
@@ -19,7 +19,7 @@ export function useUpload(url: string, errorField?: string) {
 
   /**External request for upload**/
   const requestUpload = React.useCallback(
-    (file: File) => {
+    (file: File, setUploadInProgress) => {
```

Comment:
Just a doubt - Why  setUploadInProgress need to be passed on as a parameter to the callback? Can we not handle it in FileUploader itself?

Reply:
The useEffect hook in the FileUploader has multiple dependencies. I found it more efficient to update the status of uploadInProgress in the useUpload component as soon as the upload process finishes and validate by its status in the useEffect.